UPGRADE.md

```diff
@@ -1454,6 +1454,104 @@ export const PostEdit = () => (
 
 Use the `<Title>` component instead.
 
+## `<Admin>`, `<Layout>`, `<AppBar>` And `<UserMenu>` No Longer Accept A `logout` Prop
+
+As we already provide a way to override the user menu displayed in the `<AppBar>`, we removed the `logout` prop from the `<Admin>`, `<Layout>`, `<AppBar>` and `<UserMenu>` components.
+
+If you passed your own logout component through this prop, you must now provide a custom user menu:
+
+```
+-import { Admin, Logout } from 'react-admin';
++import { Admin, AppBar, Layout, Logout, UserMenu } from 'react-admin';
+
+const MyCustomLogout = () => <Logout className="my-class-name" />;
+
++ const MyUserMenu = () => <UserMenu><MyCustomLogout /></UserMenu>;
+
++ const MyAppBar = () => <AppBar userMenu={<MyUserMenu />} />;
+
++ const MyLayout = () => <Layout appBar={MyAppBar} />;
+
+const MyAdmin = () => (
+    <Admin
+-        logout={<MyCustomLogout />}
++        layout={<MyLayout />}
+    >
+        // ....
+    </Admin>
+)
+```
+
+## `<AppBar>` and `<UserMenu>` No Longer Inject Props
+
+When a React element was provided as the `userMenu` prop, the `<AppBar>` used to clone it and inject the `logout` prop (a React element). This is no longer the case and if you provided a custom user menu, you now have to include the logout yourself.
+
+Besides, the `<UserMenu>` used to clone its children to inject the `onClick` prop, allowing them to close the menu. It now provides a `onClose` function through a new `UserContext` accessible by calling the `useUserMenu` hook.
+
+Finally, the `<UserMenu>` no longer accepts a `logout` prop. Instead, you should pass the `<Logout>` component as one of the `<UserMenu>` children.
+
+```diff
+-import { MenuItemLink, UserMenu } from 'react-admin';
++import { Logout, MenuItemLink, UserMenu, useUserMenu } from 'react-admin';
+import { Link } from 'react-router-dom';
+import MenuItem from '@mui/material/MenuItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import SettingsIcon from '@mui/icons-material/Settings';
+
+-const ConfigurationMenu = (props) => {
+// It's important to pass the ref to allow MaterialUI to manage the keyboard navigation
++const ConfigurationMenu = forwardRef((props, ref) => {
++    const { onClose } = useUserMenu();
+    return (
+        <MenuItem
```

Comment:
there should be a -MenuItemLink

Reply:
Nope, that's a MenuItem for use inside a `UserMenu` and we decided that `MenuItemLink` should only be used inside `Menu` because they toggle the Sidebar on mobile

Reply:
Sorry I misread. I still don't think so. This is how it should have been written in the first place

Reply:
The link you send is the `Menu` documentation, here we're dealing with the `UserMenu`.

However, there is also a section in the documentation for the `UserMenu` that use the `MenuItemLink` as well. I'll add a section for this too

Reply:
Done